redis/redis.go

```diff
@@ -0,0 +1,114 @@
+// OpenIO netdata collectors
+// Copyright (C) 2019 OpenIO SAS
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3.0 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public
+// License along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package redis
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+)
+
+type collector struct {
+	addr    string
+	cluster string
+}
+
+func NewCollector(addr string) *collector {
+	res := strings.Split(addr, ":")
+	if len(res) != 3 {
+		log.Fatalln("Invalid address", addr, "must be IP:PORT:CLUSTER_ID")
+	}
+	return &collector{
+		addr:    res[0] + ":" + res[1],
+		cluster: res[2],
+	}
+}
+
+var whitelist = map[string]bool{
+	"used_memory":                 true,
+	"used_memory_rss":             true,
+	"used_memory_lua":             true,
+	"mem_fragmentation_ratio":     true,
+	"rdb_changes_since_last_save": true,
+	"total_connections_received":  true,
+	"total_commands_processed":    true,
+	"instantaneous_ops_per_sec":   true,
+	"total_net_input_bytes":       true,
+	"total_net_output_bytes":      true,
+	"keyspace_hits":               true,
+	"keyspace_misses":             true,
+	"role":                        true,
+	"connected_slaves":            true,
+	"repl_backlog_size":           true,
+	"db0":                         true, // Note: VDO: maybe add support for other db?
+}
+
+var keysRegexp = regexp.MustCompile(`keys=(\d+)`)
+
+func (c *collector) Collect() (map[string]string, error) {
+	conn, err := net.Dial("tcp", c.addr)
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	_, err = conn.Write([]byte("INFO\r\nQUIT\r\n"))
+	if err != nil {
+		return nil, err
+	}
+
+	data := map[string]string{}
+
+	scanner := bufio.NewScanner(conn)
+	for scanner.Scan() {
+		line := scanner.Text()
+		kv := strings.Split(line, ":")
+		if len(kv) != 2 {
```

Comment:
Is it really ok to skip values with `:` ?

Reply:
See https://github.com/open-io/openio-netdata-plugins/pull/28/files#diff-af42133f8823b0ca58f47e0abe4da8fb. Not a single value has more than a single colon, and doesn't seem to be ab obvious way to add it via configuration on currently gathered metrics.